### PR TITLE
Bookworm

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
https://cdn.discordapp.com/attachments/660934847375867904/727280379043643492/2020-06-29_16-50-26.mp4

image
image
image

(mapping not done yet for fear of conflicts)
## Why It's Good For The Game
snek

## Changelog
:cl:
add: Bookworm, a nearsssighted sssnake pet for the curator.
tweak: Snakes now hiss when talking.
/:cl:

### CODE REVIEW

**Comment by AsciiSquid** [comment]
Ssspeed merge

**Comment by Superyodeler** [comment]
Wonderfil raptile

File: waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	icon_dead = "bookworm_dead"
+	desc = "The Curator's beloved astigmatic corn snake. Probably best not to poke it too much..."
+	speak_chance = 15 //less than poly
+	speak = list("I can't find anything to read!")
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/bookworm/Initialize()
+	glasses = new /obj/item/clothing/glasses/regular(src)
+	grant_all_languages()
+	update_overlays()
+	speak = get_phrases()
```
Comment (Superyodeler) [inline waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm:94]: It looks like this part erases "I can't find anything to read!" even if bookworm can't find anything to read.

File: waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	return
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/bookworm/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/book) && user.a_intent == INTENT_HELP)
+		var/obj/item/book/B = O
+		if(B.dat)
+			var/textdata = strip_booktext(B.dat, 25)
+			if(!(textdata in speak))
+				speak.Remove(pick(speak))
+				speak.Add("[textdata]...")
```
Comment (Superyodeler) [inline waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm:131]: The addition of the ... makes it so Bookworm won't realize if you give them the same book twice.

File: waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+		handle_automated_speech(1) //assured speak/emote
+	return
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/bookworm/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/book) && user.a_intent == INTENT_HELP)
+		var/obj/item/book/B = O
+		if(B.dat)
+			var/textdata = strip_booktext(B.dat, 25)
+			if(!(textdata in speak))
+				speak.Remove(pick(speak))
```
[the remark this answers is not in the document]
Comment (Maldaris) [reply waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm:132]: Fixed this by ensuring the speak list is never empty in a71f7a8c7e

File: waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm (hunk 1 above)
Comment (Maldaris) [reply waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm:94]: fixed in a71f7a8c7e

File: waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm (hunk 2 above)
Comment (Maldaris) [reply waspstation/code/modules/mob/living/simple_animal/friendly/snake.dm]: fixed in a71f7a8c7e

**Comment by Superyodeler** [comment]
Should probably get a testmerge

**Comment by SynnGraffkin** [comment]
Worm would need to be spawned because it isn't mapped in

**Comment by Maldaris** [comment]
Mapping can come separate now that the Map freeze with monstermos is over. Merging.

**Pull request merged**